core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -409,7 +409,8 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
 
       // If Tasks is too large, remove and garbage collect old tasks
       if (stageData.taskData.size > retainedTasks) {
-        stageData.taskData = stageData.taskData.drop(stageData.taskData.size - retainedTasks)
+        stageData.taskData = stageData.taskData.drop(
```

Comment:
Is it maybe a little clearer to write something like

```
val targetSize = 0.9 * retainedTasks
stageData.taskData = stageData.taskData.drop(stageData.taskData.size - targetSizse)
```

(And yeah does it make sense to drop more than just 1% extra, to avoid the overhead you're concerned about?)